flybywiresim/a32nx
# feat(fmgc): automatic airway intersection

Fixes #6803

## Summary of Changes

- Fix airway entry so it doesn't "remember" previous entries
- Fix airway entry anywhere other than the end of the enroute segment
- Add automatic detection of airway intersection waypoints (note, this doesn't implement geometric intersections per #5331, which is the final fallback if a waypoint intersection cannot be found)

## Screenshots (if necessary)


![image](https://user-images.githubusercontent.com/9995998/161279833-ce1c4368-7bba-47d4-b095-4d6698b97fb3.png)

## References

Honeywell pilot's guide.




## Additional context



Discord username (if different from GitHub):

## Testing instructions

- Check that you can insert successive airways without having to enter the intersection waypoint.
- Check that re-entering the airways page after insert gives you a blank page
- Check that manually adding the intersections still works okay
- Try inserting airways in the middle of an enroute segment
- Check that SimBrief INIT REQUEST still inserts airways properly


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
